sentry-android-replay/src/main/java/io/sentry/android/replay/capture/BufferCaptureStrategy.kt

```diff
@@ -228,7 +179,75 @@ internal class BufferCaptureStrategy(
                 screenAtStart = startScreen
             }
             // can clear as we switch to session mode and don't care anymore about buffering
-            bufferedSegments.clear()
+            bufferedScreens.clear()
+        }
+    }
+
+    private fun deleteFile(file: File?) {
+        if (file == null) {
+            return
+        }
+        try {
+            if (!file.delete()) {
+                options.logger.log(ERROR, "Failed to delete replay segment: %s", file.absolutePath)
+            }
+        } catch (e: Throwable) {
+            options.logger.log(ERROR, e, "Failed to delete replay segment: %s", file.absolutePath)
+        }
+    }
+
+    private fun MutableList<ReplaySegment.Created>.capture() {
+        var bufferedSegment = removeFirstOrNull()
+        while (bufferedSegment != null) {
+            bufferedSegment.capture(hub)
+            bufferedSegment = removeFirstOrNull()
+            Thread.sleep(100L)
```

Comment:
reserved for future performance optimization? 😄 In any case probably makes sense to add a comment why you're doing that.

Reply:
it's rather not to burst the server with envelopes and active rate limiting, similar to what we do here: https://github.com/getsentry/sentry-java/blob/3a89243fa0ec48fc16193070c2224833d94f9156/sentry/src/main/java/io/sentry/DirectoryProcessor.java#L103-L106

but great point about adding the comment, will do 👍